backend/app/services/diagnosis_agent.py

```
@@ -32,462 +37,552 @@ class DiagnosisState(TypedDict):
+            daily = weather_info.get("daily_forecast") if isinstance(weather_info, dict) else None
+            if daily:
+                html_cards = ["<div class='grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-3 my-3'>"]
+                for dk, td in daily.items():
+                    d_str = f"{dk[:4]}-{dk[4:6]}-{dk[6:]}"
+                    html_cards.append(f"<div class='bg-blue-50/50 border border-blue-100 rounded-xl p-3 flex flex-col items-center justify-center text-center shadow-sm'><div class='font-bold text-gray-700 mb-1 text-[13px]'>{d_str}</div><div class='font-medium text-blue-800 text-[13px] mb-1'>{td['condition']}</div><div class='text-xl flex flex-col items-center justify-center font-black text-blue-600 mb-1'><span class='text-[10px] font-normal text-gray-500 bg-white px-2 py-0.5 rounded-full shadow-sm mb-1 mt-1'>최저/최고</span><div>{td['min_temp']}°<span class='text-gray-400 text-sm font-normal mx-1'>/</span>{td['max_temp']}°</div></div><div class='flex flex-col gap-0.5 text-xs text-gray-600 mt-1'><span class='flex items-center justify-center bg-white px-2 py-0.5 rounded border border-blue-100 text-blue-800 font-medium'>☔ 강수 {td['max_precip_prob']}%</span><span class='flex items-center justify-center bg-white px-2 py-0.5 rounded border border-blue-100 text-blue-800 font-medium'>💨 풍속 {td['max_wind_speed']}m/s</span></div></div>")
+                html_cards.append("</div>")
+                weather_summary = "".join(html_cards)
+                fd = daily[sorted(daily.keys())[0]]
+                weather_for_llm = f"기온 {fd['min_temp']}~{fd['max_temp']}°C, 강수 {fd['max_precip_prob']}%"
+
+            try: grouped_results = json.loads(state.get("pesticide_data", "[]"))
```

Ruff (0.15.10)

[error] Multiple statements on one line (colon)

(E701)